### ISSUE TITLE
error due use of frame dependend ext methods

### ISSUE THREAD

**Issue by kares** [elastic/logstash#11854]
LS is calling `RubyString.gsub` from `LogStash::Util::Loggable` Java ext.
This method will attempt to set a backref for the current caller's frame.
However, when the frame stack is empty there isn't really a place to set `$~`.

```
  Exception: Java::JavaLang::ArrayIndexOutOfBoundsException
  Stack: org.jruby.runtime.ThreadContext.getCurrentFrame(ThreadContext.java:541)
org.jruby.runtime.ThreadContext.setBackRef(ThreadContext.java:565)
org.jruby.RubyString.gsubCommon(RubyString.java:3266)
org.jruby.RubyString.gsubCommon(RubyString.java:3184)
org.jruby.RubyString.gsubCommon(RubyString.java:3176)
org.jruby.RubyString.gsubImpl(RubyString.java:3167)
org.jruby.RubyString.gsub(RubyString.java:3132)
org.logstash.log.LoggableExt.log4jName(LoggableExt.java:81)
org.logstash.log.LoggableExt.access$000(LoggableExt.java:37)
org.logstash.log.LoggableExt$ClassMethods.logger(LoggableExt.java:109)
org.logstash.log.LoggableExt$ClassMethods$INVOKER$s$0$0$logger.call(LoggableExt$ClassMethods$INVOKER$s$0$0$logger.gen)
org.jruby.RubyClass.finvoke(RubyClass.java:572)
org.jruby.runtime.Helpers.invoke(Helpers.java:448)
org.jruby.RubyBasicObject.callMethod(RubyBasicObject.java:354)
org.logstash.log.LoggableExt.logger(LoggableExt.java:53)
```

### sample reproducer

*Gemfile*
```ruby
gem "logstash-input-file", :github => 'kares/logstash-input-file', :ref => 'fail-logger-error' #:path => "../plugins/logstash-input-file"
```
https://github.com/kares/logstash-input-file/commit/9c12387fd806b3a313e5d0d2e450ca04ad08c617#diff-01d90892ef2cb12636c54f203f26518eR356
(any class that `include LogStash::Util::Loggable` that will use its `logger`)

`./gradlew installDefaultGems`

`bin/logstash -e "input { file { path => '/tmp/123' } }"`

---

*HINT: if a call is nested (inside a block) the above piece will work (e.g. `1.times { SmtThatIncludesLoggable.new }` or `SmtThatIncludesLoggable.new { |block| }`)*

**Comment by kares** [comment]
based on a review of current plugins (to see how this effects 7.7.0) here's the report: filters won't be affected regardless of Java/Ruby caller since the `filter` method and friends always gets a block.

input plugins are likely only problematic so far, out of those potentially only input **snmp** is affected.
input **file** is a bit on the edge, as long as no `logger` statements is added to `FileWatch::Watch` :heavy_check_mark: 
other inputs including **jdbc** and **azure_event_hubs** only use loggable classes from within blocks.
last, the **deprecation_logger_support** mixin is affected but `deprecation_logger` isn't used so far.

**Comment by kares** [comment]
merged #11874 and 7.7 backport waiting at https://github.com/elastic/logstash/pull/11879
